test/server/active_tcp_listener_test.cc

```diff
@@ -61,6 +61,34 @@ class ActiveTcpListenerTest : public testing::Test, protected Logger::Loggable<L
   std::shared_ptr<NiceMock<Network::MockListenerFilterMatcher>> listener_filter_matcher_;
 };
 
+TEST_F(ActiveTcpListenerTest, PopulateSNIWhenNoFilterChainMatched) {
+  auto listener = std::make_unique<Network::MockListener>();
+  EXPECT_CALL(*listener, onDestroy());
+
+  NiceMock<Network::MockConnectionBalancer> balancer1;
+  Network::Address::InstanceConstSharedPtr normal_address(
+      new Network::Address::Ipv4Instance("127.0.0.1", 10001));
+  EXPECT_CALL(*socket_factory_, localAddress()).WillRepeatedly(ReturnRef(normal_address));
+  EXPECT_CALL(listener_config_, connectionBalancer()).WillRepeatedly(ReturnRef(balancer1));
+  EXPECT_CALL(listener_config_, listenerScope).Times(testing::AnyNumber());
+  EXPECT_CALL(listener_config_, listenerFiltersTimeout());
+  EXPECT_CALL(listener_config_, continueOnListenerFiltersTimeout());
+  EXPECT_CALL(listener_config_, filterChainManager()).WillRepeatedly(ReturnRef(manager_));
+  EXPECT_CALL(listener_config_, openConnections()).WillRepeatedly(ReturnRef(resource_limit_));
+
+  auto active_listener1 =
+      std::make_unique<ActiveTcpListener>(conn_handler_, std::move(listener), listener_config_);
+
+  auto accepted_socket1 = std::make_unique<NiceMock<Network::MockConnectionSocket>>();
+  EXPECT_CALL(*accepted_socket1, requestedServerName()).WillRepeatedly(Return("envoy.io"));
+
+  EXPECT_CALL(manager_, findFilterChain(_)).WillRepeatedly(Return(nullptr));
```

Comment:
I think this can be `WillOnce` instead of `WillRepeatedly`

Reply:
yes, let me fix it